test/unit/legacy_wrappers/mongo_client.test.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const { MongoClient, ChangeStream, ClientSession } = require('../../../src/index');
+const mongodbDriver = require('mongodb');
+const sinon = require('sinon');
+const { expect } = require('chai');
+
+const iLoveJs = 'mongodb://iLoveJavascript';
+
+describe('legacy_wrappers/mongo_client.js', () => {
+  let client;
+  beforeEach(async function () {
+    client = new MongoClient('mongodb://iLoveJs');
+  });
+
+  afterEach(async function () {
+    sinon.restore();
+    await client.close();
+  });
+
+  it('calling MongoClient.connect() returns promise', async () => {
+    sinon.stub(mongodbDriver.MongoClient.prototype, 'connect').returns(Promise.resolve(2));
+    expect(client).to.have.property('connect').that.is.a('function');
+    const actualReturnValue = client.connect();
+    expect(await actualReturnValue).to.be.instanceOf(MongoClient);
+  });
+
+  it('calling static MongoClient.connect(url) returns promise', async () => {
+    sinon.stub(mongodbDriver.MongoClient.prototype, 'connect').returns(Promise.resolve(2));
+    const actualReturnValue = MongoClient.connect(iLoveJs);
+    expect(await actualReturnValue).to.be.instanceOf(MongoClient);
+  });
+
+  it('calling static MongoClient.connect(url, options) returns promise', async () => {
+    sinon.stub(mongodbDriver.MongoClient.prototype, 'connect').returns(Promise.resolve(2));
+    const actualReturnValue = MongoClient.connect(iLoveJs, { appName: 'my app!' });
+    expect(await actualReturnValue).to.be.instanceOf(MongoClient);
+  });
+
+  it('calling static MongoClient.connect(url, callback)', done => {
+    sinon.stub(mongodbDriver.MongoClient.prototype, 'connect').returns(Promise.resolve(2));
+    MongoClient.connect(iLoveJs, (error, result) => {
+      try {
+        expect(error).to.be.undefined;
+        expect(result).to.be.instanceOf(MongoClient);
+        done();
+      } catch (assertionError) {
+        done(assertionError);
+      }
+    });
+  });
+
+  it('calling static MongoClient.connect(url, options, callback)', done => {
+    const returnValue = Promise.resolve(new mongodbDriver.MongoClient(iLoveJs));
+    sinon.stub(mongodbDriver.MongoClient.prototype, 'connect').returns(returnValue);
+    MongoClient.connect(iLoveJs, { appName: 'my app!' }, (error, result) => {
+      try {
+        expect(error).to.be.undefined;
+        expect(result).to.be.instanceOf(MongoClient);
+        done();
+      } catch (assertionError) {
+        done(assertionError);
+      }
+    });
+  });
+
+  it('errors thrown from MongoClient constructor are passed to the promise for static connect', async () => {
+    expect(() => new MongoClient()).to.throw; // make sure constructor still throws
```

Comment:
what is thrown from here? and what is the check to make sure that the same error is passed on?